playdate/sprite.lua

```diff
@@ -0,0 +1,468 @@
+local bit = require("bit")  -- LuaJIT's bitwise operations
+
+local module = {}
+
+module.kCollisionTypeSlide = "slide"
+module.kCollisionTypeFreeze = "freeze"
+module.kCollisionTypeOverlap = "overlap"
+module.kCollisionTypeBounce = "bounce"
+
+playdate.graphics.sprite = module
+
+local meta = {}
+meta.__index = meta
+module.__index = meta
+
+
+local allSprites = {}
+
+function module.new(imageOrTilemap)
+    local sprite = setmetatable({}, meta)
+
+    local hasSpr = sprite == nil and 'no sprite' or 'has sprite'
+
+    print("sprite " .. hasSpr)
+    -- printTable(sprite)
+
+    local hasImg = imageOrTilemap == nil and 'no image' or 'has image'
+
+    
+    print("imageOrTilemap " .. hasImg)
+    -- printTable(imageOrTilemap)
```

Comment:
Please remove this block of debugging code.

```suggestion
    local hasSpr = sprite == nil and 'no sprite' or 'has sprite'

    print("sprite " .. hasSpr)
    -- printTable(sprite)

    local hasImg = imageOrTilemap == nil and 'no image' or 'has image'

    
    print("imageOrTilemap " .. hasImg)
    -- printTable(imageOrTilemap)
```